src/main/java/com/mongodb/hibernate/internal/translate/AbstractMqlTranslator.java

```diff
@@ -229,30 +233,54 @@ public void visitColumnWriteFragment(ColumnWriteFragment columnWriteFragment) {
-    // Table Mutation: deletion
+    // Table Mutation: delete
 
     @Override
     public void visitStandardTableDelete(TableDeleteStandard tableDelete) {
         if (tableDelete.getWhereFragment() != null) {
             throw new FeatureNotSupportedException();
         }
+        var keyFilter = getKeyFilter(tableDelete);
+        astVisitorValueHolder.yield(
+                COLLECTION_MUTATION,
+                new AstDeleteCommand(tableDelete.getMutatingTable().getTableName(), keyFilter));
+    }
+
+    // ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+    // Table Mutation: update
+
+    @Override
+    public void visitStandardTableUpdate(TableUpdateStandard tableUpdate) {
+        if (tableUpdate.getWhereFragment() != null) {
+            throw new FeatureNotSupportedException();
+        }
+        var keyFilter = getKeyFilter(tableUpdate);
+        var updates = new ArrayList<AstFieldUpdate>(tableUpdate.getNumberOfValueBindings());
+        for (var valueBinding : tableUpdate.getValueBindings()) {
+            var columnExpression = valueBinding.getColumnReference().getColumnExpression();
+            var astValue = acceptAndYield(valueBinding.getValueExpression(), FIELD_VALUE);
+            updates.add(new AstFieldUpdate(columnExpression, astValue));
+        }
+        astVisitorValueHolder.yield(
+                COLLECTION_MUTATION,
+                new AstUpdateCommand(tableUpdate.getMutatingTable().getTableName(), keyFilter, updates));
+    }
 
-        if (tableDelete.getNumberOfOptimisticLockBindings() > 0) {
+    private AstFilter getKeyFilter(RestrictedTableMutation<?> tableMutation) {
+        if (tableMutation.getNumberOfOptimisticLockBindings() > 0) {
             throw new FeatureNotSupportedException("TODO-HIBERNATE-51 https://jira.mongodb.org/browse/HIBERNATE-51");
         }
 
-        if (tableDelete.getNumberOfKeyBindings() > 1) {
+        if (tableMutation.getNumberOfKeyBindings() > 1) {
             throw new FeatureNotSupportedException("MongoDB doesn't support '_id' spanning multiple columns");
         }
-        assertTrue(tableDelete.getNumberOfKeyBindings() == 1);
-        var keyBinding = tableDelete.getKeyBindings().get(0);
+        assertTrue(tableMutation.getNumberOfKeyBindings() == 1);
+        var keyBinding = tableMutation.getKeyBindings().get(0);
 
-        var tableName = tableDelete.getMutatingTable().getTableName();
         var astFilterFieldPath =
                 new AstFilterFieldPath(keyBinding.getColumnReference().getColumnExpression());
         var astValue = acceptAndYield(keyBinding.getValueExpression(), FIELD_VALUE);
-        var keyFilter = new AstFieldOperationFilter(astFilterFieldPath, new AstComparisonFilterOperation(EQ, astValue));
-        astVisitorValueHolder.yield(COLLECTION_MUTATION, new AstDeleteCommand(tableName, keyFilter));
+        return new AstFieldOperationFilter(astFilterFieldPath, new AstComparisonFilterOperation(EQ, astValue));
```

Comment:
the common id filter is refactored into its own method for it applies to both deletion and updating with no difference (both mutation classes inherits from the common `RestrictedTableMutation` class)